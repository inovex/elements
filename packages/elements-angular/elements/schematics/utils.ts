import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
