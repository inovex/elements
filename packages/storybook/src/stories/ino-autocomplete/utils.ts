import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
