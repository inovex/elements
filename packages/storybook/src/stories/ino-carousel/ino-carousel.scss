.ino-carousel-example {
  display: flex;
  flex-direction: row;
  justify-content: center;
}
