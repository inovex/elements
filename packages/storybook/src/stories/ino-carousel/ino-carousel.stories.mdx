import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-carousel/readme.md";

<Meta title="Graphic/ino-carousel/Documentation" />

<Description markdown={readme} />
