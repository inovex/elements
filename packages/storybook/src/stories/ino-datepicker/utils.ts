import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
