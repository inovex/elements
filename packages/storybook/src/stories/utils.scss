/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
