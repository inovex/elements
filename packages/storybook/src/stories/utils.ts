import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
