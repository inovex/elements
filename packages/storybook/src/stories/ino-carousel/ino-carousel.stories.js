import componentReadme from '_local-elements/src/components/ino-carousel/readme.md';
import withStencilReadme from '_local-storybookcore/with-stencil-readme';
import './ino-carousel.scss';
import { boolean, number, text } from '@storybook/addon-knobs';
import lightningImg from '_local-assets/images/lightning.jpg';
import mountainsImg from '_local-assets/images/mountains.jpg';
import nidarosImg from '_local-assets/images/nidaros.jpg';

function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function(e) {
    const el = e.target;
    if (el.tagName.toLowerCase() !== 'ino-icon-button') {
      return;
    }
    const carousel = el.closest('ino-carousel');
    const parentDiv = el.closest('div');
    if(parentDiv.classList.contains('ino-carousel__left-arrow')) {
      carousel.value = mod(carousel.value - 1, 3);
    } else {
      carousel.value = mod(carousel.value + 1, 3);
    }
  };

  const mod = (a, b) => ((a % b) + b) % b;

  document.addEventListener('clickEl', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('clickEl', eventHandler);
  };
}

export default {
  title: 'Graphic/<ino-carousel>',

  decorators: [
    withStencilReadme(componentReadme),
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

export const DefaultUsage = () => /* html */`
      <h3>Customizable carousel</h3>
      <div class="ino-carousel-example">
          <style>
            ino-carousel.customizable-carousel {
            --ino-carousel-icon-color: ${text(
        '--ino-carousel-icon-color',
        '#ffffff',
        'Custom Properties'
            )};
            --ino-carousel-width: ${text(
        '--ino-carousel-width',
        '700px',
        'Custom Properties'
            )};
            --ino-carousel-height: ${text(
        '--ino-carousel-height',
        '400px',
        'Custom Properties'
            )};
            --ino-carousel-animation-duration: ${text(
        '--ino-carousel-animation-duration',
        '700ms',
        'Custom Properties'
            )};
            }
          </style>
          <ino-carousel
            class="customizable-carousel"
            value="${text('value', '0', 'General')}"
            ino-autoplay="${boolean('ino-autoplay', false, 'General')}"
            ino-animated="${boolean('ino-animated', false, 'General')}"
            ino-hide-buttons="${boolean('ino-hide-buttons', false, 'General')}"
            ino-infinite="${boolean('ino-infinite', false, 'General')}"
            ino-intermission="${number('ino-intermission', 5000, undefined, 'General')}"
            ino-reverse="${boolean('ino-reverse', false, 'General')}"
          >
            <ino-carousel-slide src=${lightningImg} value="0" ino-selected>
            </ino-carousel-slide>
            <ino-carousel-slide src=${mountainsImg} value="1">
            </ino-carousel-slide>
            <ino-carousel-slide src=${nidarosImg} value="2">
            </ino-carousel-slide>
          </ino-carousel>
      </div>

      <h3>Autoplay enabled</h3>
      <ino-carousel ino-autoplay ino-animated ino-infinite value="0">
            <ino-carousel-slide src=${lightningImg} value="0">
            </ino-carousel-slide>
            <ino-carousel-slide src=${mountainsImg} value="1">
            </ino-carousel-slide>
            <ino-carousel-slide src=${nidarosImg} value="2">
            </ino-carousel-slide>
      </ino-carousel>
`;
