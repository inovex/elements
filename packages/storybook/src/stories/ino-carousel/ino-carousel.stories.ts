import { Components } from '@inovex.de/elements';
import { useEffect } from '@storybook/client-api';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';

// @ts-ignore
import lightningImg from '../../assets/images/lightning.jpg';
// @ts-ignore
import mountainsImg from '../../assets/images/mountains.jpg';
// @ts-ignore
import nidarosImg from '../../assets/images/nidaros.jpg';

import { defaultDecorator } from '../utils';

import './ino-carousel.scss';

export default {
  title: 'Graphic/<ino-carousel>',
  component: 'ino-carousel',
  decorators: [
    defaultDecorator,
    (story) => {
      useEffect(() => {
        const eventHandler = function (e: any) {
          if (e.target.tagName.toLowerCase() !== 'ino-icon-button') {
            return;
          }
          const iconButton: HTMLInoIconButtonElement = e.target;

          const carousel: HTMLInoCarouselElement = iconButton.closest(
            'ino-carousel'
          );
          const isLeftArrow = iconButton.icon === 'arrow_left';
          carousel.value = mod(carousel.value + (isLeftArrow ? -1 : 1), 3);
        };

        const mod = (a, b) => ((a % b) + b) % b;

        document.addEventListener('clickEl', eventHandler);

        return () => {
          document.removeEventListener('clickEl', eventHandler);
        };
      });
      return story();
    },
  ],
};

export const Playground: Story<Components.InoCarousel> = (args) => html`
  <h3>Customizable carousel</h3>
  <div class="ino-carousel-example">
    <ino-carousel
      class="customizable-carousel"
      value="${args.value}"
      autoplay="${args.autoplay}"
      animated="${args.animated}"
      hide-buttons="${args.hideButtons}"
      infinite="${args.infinite}"
      intermission="${args.intermission}"
      reverse="${args.reverse}"
    >
      <ino-carousel-slide src=${lightningImg} value="0"> </ino-carousel-slide>
      <ino-carousel-slide src=${mountainsImg} value="1"></ino-carousel-slide>
      <ino-carousel-slide src=${nidarosImg} value="2"></ino-carousel-slide>
    </ino-carousel>
  </div>
`;

Playground.args = {
  value: 0,
  autoplay: false,
  animated: false,
  hideButtons: false,
  infinite: false,
  intermission: 5000,
  reverse: false,
};

export const Autoplay = () => html`
  <ino-carousel intermission="2000" autoplay animated infinite value="0">
    <ino-carousel-slide src=${lightningImg} value="0"></ino-carousel-slide>
    <ino-carousel-slide src=${mountainsImg} value="1"></ino-carousel-slide>
    <ino-carousel-slide src=${nidarosImg} value="2"></ino-carousel-slide>
  </ino-carousel>
`;
