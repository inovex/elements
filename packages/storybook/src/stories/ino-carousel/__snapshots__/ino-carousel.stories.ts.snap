// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graphic/<ino-carousel> Animated smoke-test 1`] = `
<div id="root-inner">
  <div class="story-carousel">
    <div class="ino-carousel-example">
      <ino-carousel class="customizable-carousel hydrated"
                    value="0"
                    autoplay="false"
                    animated="true"
                    hide-buttons="false"
                    infinite="true"
                    intermission="5000"
                    reverse="false"
      >
        <div class="ino-carousel">
          <div class="ino-carousel__container">
            <ino-carousel-slide value="0"
                                src="static/media/src/assets/images/lightning.jpg"
                                class="hydrated ino-carousel-slide--selected"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/lightning.jpg&quot;);">
              </div>
            </ino-carousel-slide>
            <ino-carousel-slide value="1"
                                src="static/media/src/assets/images/mountains.jpg"
                                class="hydrated"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/mountains.jpg&quot;);">
              </div>
            </ino-carousel-slide>
            <ino-carousel-slide value="2"
                                src="static/media/src/assets/images/nidaros.jpg"
                                class="hydrated"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/nidaros.jpg&quot;);">
              </div>
            </ino-carousel-slide>
          </div>
          <div class="ino-carousel__left-arrow">
            <ino-icon-button class="ino-icon-button--color-scheme-primary hydrated">
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded"
                      type="button"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="ino-icon--color-scheme-primary hydrated">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
          <div class="ino-carousel__right-arrow">
            <ino-icon-button class="ino-icon-button--color-scheme-primary hydrated">
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded"
                      type="button"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="ino-icon--color-scheme-primary hydrated">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
        </div>
      </ino-carousel>
    </div>
  </div>
</div>
`;

exports[`Graphic/<ino-carousel> Autoplay smoke-test 1`] = `
<div id="root-inner">
  <div class="story-carousel">
    <div class="ino-carousel-example">
      <ino-carousel class="customizable-carousel hydrated"
                    value="0"
                    autoplay="true"
                    animated="true"
                    hide-buttons="false"
                    infinite="true"
                    intermission="2000"
                    reverse="false"
      >
        <div class="ino-carousel">
          <div class="ino-carousel__container">
            <ino-carousel-slide value="0"
                                src="static/media/src/assets/images/lightning.jpg"
                                class="hydrated ino-carousel-slide--selected"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/lightning.jpg&quot;);">
              </div>
            </ino-carousel-slide>
            <ino-carousel-slide value="1"
                                src="static/media/src/assets/images/mountains.jpg"
                                class="hydrated"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/mountains.jpg&quot;);">
              </div>
            </ino-carousel-slide>
            <ino-carousel-slide value="2"
                                src="static/media/src/assets/images/nidaros.jpg"
                                class="hydrated"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/nidaros.jpg&quot;);">
              </div>
            </ino-carousel-slide>
          </div>
          <div class="ino-carousel__left-arrow">
            <ino-icon-button class="ino-icon-button--color-scheme-primary hydrated">
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                      type="button"
                      style="--mdc-ripple-fg-size:28px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:10px; --mdc-ripple-top:10px;"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="ino-icon--color-scheme-primary hydrated">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
          <div class="ino-carousel__right-arrow">
            <ino-icon-button class="ino-icon-button--color-scheme-primary hydrated">
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                      type="button"
                      style="--mdc-ripple-fg-size:28px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:10px; --mdc-ripple-top:10px;"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="ino-icon--color-scheme-primary hydrated">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
        </div>
      </ino-carousel>
    </div>
  </div>
</div>
`;

exports[`Graphic/<ino-carousel> HideButtons smoke-test 1`] = `
<div id="root-inner">
  <div class="story-carousel">
    <div class="ino-carousel-example">
      <ino-carousel class="customizable-carousel hydrated"
                    value="0"
                    autoplay="false"
                    animated="false"
                    hide-buttons="true"
                    infinite="true"
                    intermission="5000"
                    reverse="false"
      >
        <div class="ino-carousel ino-carousel--no-buttons">
          <div class="ino-carousel__container">
            <ino-carousel-slide value="0"
                                src="static/media/src/assets/images/lightning.jpg"
                                class="hydrated ino-carousel-slide--selected"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/lightning.jpg&quot;);">
              </div>
            </ino-carousel-slide>
            <ino-carousel-slide value="1"
                                src="static/media/src/assets/images/mountains.jpg"
                                class="hydrated"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/mountains.jpg&quot;);">
              </div>
            </ino-carousel-slide>
            <ino-carousel-slide value="2"
                                src="static/media/src/assets/images/nidaros.jpg"
                                class="hydrated"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/nidaros.jpg&quot;);">
              </div>
            </ino-carousel-slide>
          </div>
          <div class="ino-carousel__left-arrow">
            <ino-icon-button class="ino-icon-button--color-scheme-primary hydrated">
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded"
                      type="button"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="ino-icon--color-scheme-primary hydrated">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
          <div class="ino-carousel__right-arrow">
            <ino-icon-button class="ino-icon-button--color-scheme-primary hydrated">
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded"
                      type="button"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="ino-icon--color-scheme-primary hydrated">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
        </div>
      </ino-carousel>
    </div>
  </div>
</div>
`;

exports[`Graphic/<ino-carousel> Infinite smoke-test 1`] = `
<div id="root-inner">
  <div class="story-carousel">
    <div class="ino-carousel-example">
      <ino-carousel class="customizable-carousel"
                    value="0"
                    autoplay="false"
                    animated="false"
                    hide-buttons="false"
                    infinite="true"
                    intermission="5000"
                    reverse="false"
      >
        <ino-carousel-slide value="0"
                            src="static/media/src/assets/images/lightning.jpg"
        >
        </ino-carousel-slide>
        <ino-carousel-slide value="1"
                            src="static/media/src/assets/images/mountains.jpg"
        >
        </ino-carousel-slide>
        <ino-carousel-slide value="2"
                            src="static/media/src/assets/images/nidaros.jpg"
        >
        </ino-carousel-slide>
      </ino-carousel>
    </div>
  </div>
</div>
`;

exports[`Graphic/<ino-carousel> Intermission smoke-test 1`] = `
<div id="root-inner">
  <div class="story-carousel">
    <div class="ino-carousel-example">
      <ino-carousel class="customizable-carousel hydrated"
                    value="0"
                    autoplay="true"
                    animated="true"
                    hide-buttons="false"
                    infinite="true"
                    intermission="2000"
                    reverse="false"
      >
        <div class="ino-carousel">
          <div class="ino-carousel__container">
            <ino-carousel-slide value="0"
                                src="static/media/src/assets/images/lightning.jpg"
                                class="hydrated ino-carousel-slide--selected"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/lightning.jpg&quot;);">
              </div>
            </ino-carousel-slide>
            <ino-carousel-slide value="1"
                                src="static/media/src/assets/images/mountains.jpg"
                                class="hydrated"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/mountains.jpg&quot;);">
              </div>
            </ino-carousel-slide>
            <ino-carousel-slide value="2"
                                src="static/media/src/assets/images/nidaros.jpg"
                                class="hydrated"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/nidaros.jpg&quot;);">
              </div>
            </ino-carousel-slide>
          </div>
          <div class="ino-carousel__left-arrow">
            <ino-icon-button class="ino-icon-button--color-scheme-primary hydrated">
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                      type="button"
                      style="--mdc-ripple-fg-size:28px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:10px; --mdc-ripple-top:10px;"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="ino-icon--color-scheme-primary hydrated">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
          <div class="ino-carousel__right-arrow">
            <ino-icon-button class="ino-icon-button--color-scheme-primary hydrated">
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                      type="button"
                      style="--mdc-ripple-fg-size:28px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:10px; --mdc-ripple-top:10px;"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="ino-icon--color-scheme-primary hydrated">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
        </div>
      </ino-carousel>
    </div>
  </div>
</div>
`;

exports[`Graphic/<ino-carousel> Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-carousel">
    <div class="ino-carousel-example">
      <ino-carousel class="customizable-carousel hydrated"
                    value="0"
                    autoplay="false"
                    animated="false"
                    hide-buttons="false"
                    infinite="true"
                    intermission="5000"
                    reverse="false"
      >
        <div class="ino-carousel">
          <div class="ino-carousel__container">
            <ino-carousel-slide value="0"
                                src="static/media/src/assets/images/lightning.jpg"
                                class="hydrated ino-carousel-slide--selected"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/lightning.jpg&quot;);">
              </div>
            </ino-carousel-slide>
            <ino-carousel-slide value="1"
                                src="static/media/src/assets/images/mountains.jpg"
                                class="hydrated"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/mountains.jpg&quot;);">
              </div>
            </ino-carousel-slide>
            <ino-carousel-slide value="2"
                                src="static/media/src/assets/images/nidaros.jpg"
                                class="hydrated"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/nidaros.jpg&quot;);">
              </div>
            </ino-carousel-slide>
          </div>
          <div class="ino-carousel__left-arrow">
            <ino-icon-button class="ino-icon-button--color-scheme-primary hydrated">
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                      type="button"
                      style="--mdc-ripple-fg-size:28px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:10px; --mdc-ripple-top:10px;"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="hydrated ino-icon--color-scheme-primary">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
          <div class="ino-carousel__right-arrow">
            <ino-icon-button class="ino-icon-button--color-scheme-primary hydrated">
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                      type="button"
                      style="--mdc-ripple-fg-size:28px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:10px; --mdc-ripple-top:10px;"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="hydrated ino-icon--color-scheme-primary">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
        </div>
      </ino-carousel>
    </div>
  </div>
</div>
`;

exports[`Graphic/<ino-carousel> Reverse smoke-test 1`] = `
<div id="root-inner">
  <div class="story-carousel">
    <div class="ino-carousel-example">
      <ino-carousel class="customizable-carousel hydrated"
                    value="0"
                    autoplay="true"
                    animated="true"
                    hide-buttons="false"
                    infinite="true"
                    intermission="2000"
                    reverse="true"
      >
        <div class="ino-carousel">
          <div class="ino-carousel__container">
            <ino-carousel-slide value="0"
                                src="static/media/src/assets/images/lightning.jpg"
                                class="hydrated ino-carousel-slide--selected"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/lightning.jpg&quot;);">
              </div>
            </ino-carousel-slide>
            <ino-carousel-slide value="1"
                                src="static/media/src/assets/images/mountains.jpg"
                                class="hydrated"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/mountains.jpg&quot;);">
              </div>
            </ino-carousel-slide>
            <ino-carousel-slide value="2"
                                src="static/media/src/assets/images/nidaros.jpg"
                                class="hydrated"
            >
              <div style="background-image: url(&quot;static/media/src/assets/images/nidaros.jpg&quot;);">
              </div>
            </ino-carousel-slide>
          </div>
          <div class="ino-carousel__left-arrow">
            <ino-icon-button class="ino-icon-button--color-scheme-primary hydrated">
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded"
                      type="button"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="ino-icon--color-scheme-primary hydrated">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
          <div class="ino-carousel__right-arrow">
            <ino-icon-button class="ino-icon-button--color-scheme-primary hydrated">
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded"
                      type="button"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="ino-icon--color-scheme-primary hydrated">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
        </div>
      </ino-carousel>
    </div>
  </div>
</div>
`;
